src/Native/System.Security.Cryptography.Native/pal_x509.cpp

```diff
@@ -33,3 +73,182 @@ extern "C" X509* DecodeX509(const unsigned char* buf, int32_t len)
+{
+    return i2d_X509(x, nullptr);
+}
+
+extern "C" int32_t EncodeX509(X509* x, uint8_t* buf)
+{
+    return i2d_X509(x, &buf);
+}
+
+extern "C" void X509Destroy(X509* a)
+{
+    if (a != nullptr)
+    {
+        X509_free(a);
+    }
+}
+
+extern "C" X509* X509Duplicate(X509* x509)
+{
+    return X509_dup(x509);
+}
+
+extern "C" X509* PemReadX509FromBio(BIO* bio)
+{
+    return PEM_read_bio_X509_AUX(bio, nullptr, nullptr, nullptr);
+}
+
+extern "C" ASN1_INTEGER* X509GetSerialNumber(X509* x509)
+{
+    return X509_get_serialNumber(x509);
+}
+
+extern "C" X509_NAME* X509GetIssuerName(X509* x509)
+{
+    return X509_get_issuer_name(x509);
+}
+
+extern "C" X509_NAME* X509GetSubjectName(X509* x509)
+{
+    return X509_get_subject_name(x509);
+}
+
+extern "C" int32_t X509CheckPurpose(X509* x, int32_t id, int32_t ca)
+{
+    return X509_check_purpose(x, id, ca);
+}
+
+extern "C" int32_t X509CheckIssued(X509* issuer, X509* subject)
+{
+    return X509_check_issued(issuer, subject);
+}
+
+extern "C" uint64_t X509IssuerNameHash(X509* x)
+{
+    return X509_issuer_name_hash(x);
+}
+
+extern "C" int32_t X509GetExtCount(X509* x)
+{
+    return X509_get_ext_count(x);
+}
+
+extern "C" X509_EXTENSION* X509GetExt(X509* x, int32_t loc)
+{
+    return X509_get_ext(x, loc);
+}
+
+extern "C" ASN1_OBJECT* X509ExtensionGetObject(X509_EXTENSION* x)
+{
+    return X509_EXTENSION_get_object(x);
+}
+
+extern "C" ASN1_OCTET_STRING* X509ExtensionGetData(X509_EXTENSION* x)
+{
+    return X509_EXTENSION_get_data(x);
+}
+
+extern "C" int32_t X509ExtensionGetCritical(X509_EXTENSION* x)
+{
+    return X509_EXTENSION_get_critical(x);
+}
+
+extern "C" X509_STORE* X509StoreCreate()
+{
+    return X509_STORE_new();
+}
+
+extern "C" void X509StoreDestory(X509_STORE* v)
+{
+    if (v != nullptr)
+    {
+        X509_STORE_free(v);
+    }
+}
+
+extern "C" int32_t X509StoreAddCert(X509_STORE* ctx, X509* x)
+{
+    return X509_STORE_add_cert(ctx, x);
+}
+
+extern "C" int32_t X509StoreAddCrl(X509_STORE* ctx, X509_CRL* x)
+{
+    return X509_STORE_add_crl(ctx, x);
+}
+
+extern "C" int32_t X509StoreSetRevocationFlag(X509_STORE* ctx, X509RevocationFlag revocationFlag)
+{
+    unsigned long verifyFlags = X509_V_FLAG_CRL_CHECK;
+
+    if (revocationFlag != X509RevocationFlag::EndCertificateOnly)
+    {
+        verifyFlags |= X509_V_FLAG_CRL_CHECK_ALL;
+    }
+
+    return X509_STORE_set_flags(ctx, verifyFlags);
+}
+
+extern "C" X509_STORE_CTX* X509StoreCtxCreate()
+{
+    return X509_STORE_CTX_new();
+}
+
+extern "C" void X509StoreCtxDestroy(X509_STORE_CTX* v)
+{
+    if (v != nullptr)
+    {
+        X509_STORE_CTX_free(v);
+    }
+}
+
+extern "C" int32_t X509StoreCtxInit(X509_STORE_CTX* ctx, X509_STORE* store, X509* x509)
+{
+    return X509_STORE_CTX_init(ctx, store, x509, nullptr);
+}
+
+extern "C" int32_t X509VerifyCert(X509_STORE_CTX* ctx)
+{
+    return X509_verify_cert(ctx);
+}
+
+extern "C" X509Stack* X509StoreCtxGet1Chain(X509_STORE_CTX* ctx)
```

Comment:
I'd leave the "1" out of the name, personally.  It means it's the 1 argument get; but there isn't any other get.  Sort of like calling a function SomeFunctionE, in anticipation of putting an x on the end :)